seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ConnectorPackageClient.java

```diff
@@ -0,0 +1,149 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.engine.client.job;
+
+import org.apache.seatunnel.engine.client.SeaTunnelHazelcastClient;
+import org.apache.seatunnel.engine.common.utils.MDUtil;
+import org.apache.seatunnel.engine.core.job.ConnectorJar;
+import org.apache.seatunnel.engine.core.job.ConnectorJarIdentifier;
+import org.apache.seatunnel.engine.core.job.ConnectorJarType;
+import org.apache.seatunnel.engine.core.protocol.codec.SeaTunnelUploadConnectorJarCodec;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.apache.seatunnel.shade.com.google.common.base.Preconditions.checkNotNull;
+
+public class ConnectorPackageClient {
+
+    private static final ILogger LOGGER = Logger.getLogger(ConnectorPackageClient.class);
+
+    private final SeaTunnelHazelcastClient hazelcastClient;
+
+    public ConnectorPackageClient(SeaTunnelHazelcastClient hazelcastClient) {
+        checkNotNull(hazelcastClient);
+        this.hazelcastClient = hazelcastClient;
+    }
+
+    public Set<ConnectorJarIdentifier> uploadCommonPluginJars(
+            long jobId, List<URL> commonPluginJars) {
+        Set<ConnectorJarIdentifier> connectorJarIdentifiers = new HashSet<>();
+        // Upload commonPluginJar
+        for (URL commonPluginJar : commonPluginJars) {
+            // handle the local file path
+            // origin path : /${SEATUNNEL_HOME}/plugins/Jdbc/lib/mysql-connector-java-5.1.32.jar ->
+            // handled path : ${SEATUNNEL_HOME}/plugins/Jdbc/lib/mysql-connector-java-5.1.32.jar
+            Path path = Paths.get(commonPluginJar.getPath().substring(1));
+            // Obtain the directory name of the relative location of the file path.
+            // for example, The path is
+            // ${SEATUNNEL_HOME}/plugins/Jdbc/lib/mysql-connector-java-5.1.32.jar, so the name
+            // obtained here is the connector plugin name : JDBC
```

Comment:
The logic not right. In fact, we don't must use plugin name like `Jdbc` to create directory. It could be any name, like `today`, `mysql` etc. Directory name doesn't have any relation with plugin name. The jar in `plugin/` should used for all connector. (PS: in Zeta engine, we mark `plugin` directory as deprecated. Refer: https://github.com/apache/seatunnel/blob/dev/plugins/README.md, user should put jar into `lib` on all node. I think in this situation, you should not care common jar). But we supported define common jar in job env. Refer https://seatunnel.apache.org/docs/2.3.3/concept/JobEnvConfig#jars

Reply:
May I ask if we can keep the function of uploading common jars to the Zeta engine？ When common jars do not exist in Zeta's `lib`, it can upload the local common jars of the client to the lib directory of all engine nodes. This way, even if the user does not place a jar on all nodes in Zeta's lib, the task can still be executed normally.

Reply:
> May I ask if we can keep the function of uploading common jars to the Zeta engine？

Yes, keep it. But marked as not recommend.

> When common jars do not exist in Zeta's lib, it can upload the local common jars of the client to the lib directory of all engine nodes.

Do not touch `lib` directory. This may cause exceptions after the server restarts. If user put jar in `plugin/`, just upload to `plugin/` in server.

Reply:
So. You still don't fix this problem.

Reply:
I'm very sorry, I don't quite understand your previous description. I have made modifications to this part of the code. Could you please help me see if this logic is right? If not, I will make the necessary modifications.